src/admin/class-datalayers.php

```diff
@@ -116,28 +167,67 @@ public static function add_datalayer_metaboxes() {
 					'url' => false,
 				],
 				'attributes' => [
-					'data-validation' => 'required',
+					'data-conditional-id'    => 'datalayer_type',
+					'data-conditional-value' => 'fileinput',
 				],
 			]
 		);
 
-		$cmb = new_cmb2_box(
+		$cmb->add_field(
 			[
-				'id'           => 'title_field_mapping_metabox',
-				'title'        => __( 'Title field mapping', 'openkaarten-base' ),
-				'object_types' => [ 'owc_ok_datalayer' ],
-				'show_on_cb'   => [ 'Openkaarten_Base_Plugin\Admin\Datalayers', 'show_field_mapping_metabox' ],
+				'name'       => __( 'Datalayer URL', 'openkaarten-base' ),
+				'id'         => 'datalayer_url',
+				'type'       => 'text_url',
+				'desc'       => __( 'Insert a valid URL. The URL must be accessible by the server and the output has to be a supported format: JSON, geoJSON, KML or XML.', 'openkaarten-base' ),
+				'attributes' => [
+					'data-conditional-id'    => 'datalayer_type',
+					'data-conditional-value' => 'url',
+				],
 			]
 		);
 
-		$source_fields = self::cmb2_get_source_fields( $cmb->object_id() );
+		if ( ! self::$cmb_object_id ) {
+			return;
+		}
+
+		self::$datalayer_type = get_post_meta( self::$cmb_object_id, 'datalayer_type', true );
+
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Nonce verification is not needed here.
```

Comment:
Waarom zou nonce verificatie hier niet nodig zijn? We hebben in principe te maken met user input